File: src/Lession4/Ex2.java
package Lession4;

import java.util.Scanner;

public class Ex2 {
    static void main() {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter a string: ");
        String str = sc.nextLine();
        System.out.println(reverseWords(str));
    }

    static String reverseWords (String inputString) {
        StringBuilder sb = new StringBuilder();
        int end = inputString.length() - 1;
        for (;;) {
            int start = inputString.lastIndexOf(' ', end);
            for (int k = start + 1; k <= end; k++) {
                sb.append(inputString.charAt(k));
            }
            if (start < 0) {
                break;
            }
            sb.append(' ');
            end = start - 1;
        }
        return sb.toString();
    }
}

File: src/Lession4/Ex2.kt
package Lession4

fun main() {
    print("Enter a string: ")
    val string = readln()
    print(reverseWords(string))

}

fun reverseWords(inputString: String): String {
    val sb = StringBuilder()
    var end = inputString.length - 1
    while(true) {
        val start = inputString.lastIndexOf(' ', end)
        for (k in start + 1 .. end) {
            sb.append(inputString[k])
        }
        if (start < 0) {
            break;
        }
        sb.append(' ')
        end = start - 1
    }
    return sb.toString()
}
